### PR TITLE
Capability improvements to enable TDS

### PR DESCRIPTION
**Pull request by simopier**
This PR improves current TMAP8 to support thermal desorption spectroscopy (TDS) modeling. 

The commit messages describe what is being done. 

(Ref. #133) - a follow up PR will add an example case for TDS to complete this issue, but this PR already grew large enough.. 
(Close. #134)

### CODE REVIEW

**Comment by simopier** [comment]
I created #136 to potentially add deprecated tests in CIVET recipes since it would be nice to run them here.

**Comment by cticenhour** [comment]
@simopier should have time to give this a full review tomorrow.

File: src/nodal_kernels/TrappingNodalKernel.C [hunk 1]
```diff
@@ -17,31 +18,35 @@ TrappingNodalKernel::validParams()
...
+      "Ct0", "The fraction of host sites that can contribute to trapping as a function (-)");
   params.addParam<Real>(
       "trap_per_free",
       1.,
       "An estimate for the ratio of the concentration magnitude of trapped species to free "
-      "species. Setting a value for this can be helpful in producing a well-scaled matrix");
+      "species. Setting a value for this can be helpful in producing a well-scaled matrix (-)");
   params.addRequiredCoupledVar(
-      "mobile", "The variable representing the mobile concentration of solute particles");
+      "mobile", "The variable representing the mobile concentration of solute particles (1/m^3)");
```
Comment (cticenhour) [inline src/nodal_kernels/TrappingNodalKernel.C:33]: I think this is a bit too vague. `mobile_concentration` instead?

File: test/tests/bcs/EquilibriumBC/equilibriumBC.i [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  dtmin = .1
+  l_max_its = 30
+  nl_max_its = 5
+  petsc_options = '-snes_converged_reason -ksp_monitor_true_residual'
+  petsc_options_iname = '-pc_type -mat_mffd_err'
+  petsc_options_value = 'lu       1e-5'
+  line_search = 'bt'
+  scheme = 'bdf2'
+  timestep_tolerance = 1e-8
+[]
```
Comment (cticenhour) [inline test/tests/bcs/EquilibriumBC/equilibriumBC.i:73]: This seems to be a simple test. Can this executioner block be simplified? I would guess so.

File: test/tests/bcs/EquilibriumBC/tests [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  [equilibriumBC_fiel_variable]
```
Comment (cticenhour) [inline test/tests/bcs/EquilibriumBC/tests:2]:
```suggestion
  [equilibriumBC_field_variable]
```

**Review by cticenhour — CHANGES_REQUESTED** [review]

File: test/tests/bcs/EquilibriumBC/tests (hunk 3 above)
Comment (simopier) [reply test/tests/bcs/EquilibriumBC/tests:2]: Great catch!

File: test/tests/bcs/EquilibriumBC/tests [hunk 4]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  [equilibriumBC_field_variable]
+    type = 'CSVDiff'
+    input = 'equilibriumBC.i'
+    csvdiff = 'equilibriumBC_out.csv'
+    issues = '#134'
+    design = 'EquilibriumBC.md'
+    requirement = 'The system shall compute the flux imposed by EquilibriumBC when a field variable is used for the enclosure variable.'
```
Comment (cticenhour) [inline test/tests/bcs/EquilibriumBC/tests:8]:
```suggestion
    requirement = 'The system shall compute the equilibrium flux condition between an enclosure and a structure when a field variable is used for the enclosure variable.'
```

On second thought, let's not use the object name. The object and its usage should _ideally_ arise from the requirement, not the other way around.

**Review by cticenhour — CHANGES_REQUESTED** [review]
A couple more items that were missed / reconsidered. I think this is ready to go after these are addressed.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**